### ISSUE TITLE
Infinite loop on NtfsAttributes custom iterator next

### ISSUE THREAD

**Issue by Arthur-PI** [ColinFinck/ntfs#22]
There is bug which lead to an infinite loop.
In the `next` implemention of `Iterator` in `NtfsAttrubutes` here:
https://github.com/ColinFinck/ntfs/blob/9348d729bcefe0b9473d29b8d7972d5e4038c910/src/attribute.rs#L508-L583
There is a case where next while never return `None` indicating the end. I searched but I'm not able to find the problem and how to solve the issue.
The infinite call is here:
https://github.com/ColinFinck/ntfs/blob/9348d729bcefe0b9473d29b8d7972d5e4038c910/src/file.rs#L216-L231
`iter.next()` is never returning `None` so it never ends.
Here is the file to reproduce the bug [test48.zip](https://github.com/ColinFinck/ntfs/files/10352611/test48.zip)

**Comment by Arthur-PI** [comment]
News! After some digging, turns out the problem is in the seek implementation of `NtfsNonResidentAttributeValue` here :
https://github.com/ColinFinck/ntfs/blob/9348d729bcefe0b9473d29b8d7972d5e4038c910/src/attribute_value/non_resident.rs#L149-L193
And the problem is that `bytes_left_to_seek` is equal it 0 and so it does not seek a thing, so the state of `NonResidentAttributeValue` never change and so on...  
So I have implemented a simple fix once again but I doubt that this is is the preferred solution for you, you tell me.
So my solution is to just add before the `while bytes_left_to_seek > 0`
```rust
if bytes_left_to_seek == 0 {
    return Err(NtfsError::NoBytesToSeekNonResidentValue {});
}
```
With a new custom error that need to be decide

**Comment by ColinFinck** [comment]
Hey @Arthur-PI! Thanks for opening this bug report.
As you already saw, I fixed it a bit differently in #23, but an explanation is still missing. So here it comes: Zero-byte seeks are totally legit and also happen very often when parsing healthy filesystems. In fact, `.seek(SeekFrom::Current(0))` has been the official way to determine the current seek position in Rust for a long time until they introduced `.stream_position()` in 1.51.0.
As a result, we can't fix this problem by treating zero-byte seeks as an error.

Instead, I've added code to reset `stream_data_run` and `data_position` when seeking beyond the data covered by Data Runs (for non-resident attributes). Previously, attributes could lie about their total length, and your `test48` file did exactly that (`u64::MAX` was chosen for length). This also means that the "infinite loop" was finite after all, just taking VERY long until all Data Runs were seeked (and working with a stale `data_position` in the process -- definitely a serious bug!). The new code bails out early now as soon as we seeked through all available Data Runs and none is left.

Your `test48` file also has a Data Run with zero `cluster_count` but non-zero `vcn`, which is also invalid. I've added a validation step to also return a specific error in this case.

**Comment by ColinFinck** [comment]
Fixed in #23
